diff -ur src/ekg.c src-patch/ekg.c
--- src/ekg.c	Tue Feb 26 18:48:48 2002
+++ src-patch/ekg.c	Mon Mar  4 16:16:44 2002
@@ -98,7 +98,7 @@
 		for (l = watches; l; l = l->next) {
 			struct gg_common *w = l->data;
 
-			if (!w || w->state == GG_STATE_ERROR || w->state == GG_STATE_IDLE || w->state == GG_STATE_DONE)
+			if (!l->data || w->state == GG_STATE_ERROR || w->state == GG_STATE_IDLE || w->state == GG_STATE_DONE)
 				continue;
 			
 			if (w->fd > maxfd)
@@ -123,6 +123,9 @@
 				struct gg_dcc *d = l->data;
 				char *errmsg = "";
 
+				if (!l->data)
+					continue;
+
 				if (c->timeout == -1)
 					continue;
 
@@ -237,7 +240,7 @@
 
 					m = l->next;
 
-					if (pid != p->pid)
+					if (!l->data || pid != p->pid)
 						continue;
 
 					if (p->name[0] == '\001') {
@@ -258,7 +261,7 @@
 				struct gg_common *c = l->data;
 				int i;
 
-				if (!FD_ISSET(c->fd, &rd) && !FD_ISSET(c->fd, &wd))
+				if (!l->data || !FD_ISSET(c->fd, &rd) && !FD_ISSET(c->fd, &wd))
 					continue;
 
 				if (c->type == GG_SESSION_USER0) 
