diff -ur src/commands.c src-patch/commands.c
--- src/commands.c	Thu Feb 21 18:42:34 2002
+++ src-patch/commands.c	Mon Mar  4 19:06:23 2002
@@ -1078,7 +1078,21 @@
 	}
 
 	/* list --active | --busy | --inactive */
-        if (params[0] && (argv = array_make(params[0], " \t", 0, 1, 1))) {
+	if (params[1])
+		tmp = (char *)malloc(strlen(params[0]) + strlen(params[1]) + 1);
+	else
+		tmp = (char *)malloc(strlen(params[0]));
+	if (!tmp)
+		return 0;
+
+	strcpy(tmp, params[0]);
+	if (params[1])
+	{
+		strcat(tmp, " ");
+		strcat(tmp, params[1]);
+	}
+
+	if (argv = array_make(tmp, " \t", 0, 1, 1)) {
 		int i;
 
 		for (i = 0; argv[i]; i++) {
@@ -1101,6 +1115,7 @@
 			}
 		}
 	}
+	free(tmp);
 
 	for (l = userlist; l; l = l->next) {
 		struct userlist *u = l->data;
