ascent@linux.pl


diff -Naur ekg-2001100103/configure ekg-2001100103.work/configure
--- ekg-2001100103/configure	Mon Oct  1 13:15:15 2001
+++ ekg-2001100103.work/configure	Mon Oct  1 14:25:14 2001
@@ -133,21 +133,17 @@
 if ! $cc $tmpfile_c -pipe -o $tmpfile > /dev/null 2>&1; then
   rm -f $tmpfile $tmpfile_c
   echo "nie."
-  exit 1
+else 
+ $tmpfile >/dev/null 2>&1
+ if [ "$?" != "7" ]; then
+   rm -f $tmpfile $tmpfile_c
+   echo "jakos dziwnie??"
+ else 
+   rm -f $tmpfile $tmpfile_c
+   cflags="$cflags -pipe"
+   echo "taaak."
+ fi 
 fi
-
-$tmpfile
-
-if [ "$?" != "7" ]; then
-  rm -f $tmpfile $tmpfile_c
-  echo "jakos dziwnie??"
-  exit 1
-fi
-
-rm -f $tmpfile $tmpfile_c
-cflags="$cflags -pipe"
-
-echo "taaak."
 
 #
 # endiany.
