diff -r -C 2 ekg-20020414/docs/vars.txt ekg-20020414-new/docs/vars.txt
*** ekg-20020414/docs/vars.txt	Thu Mar 28 13:40:45 2002
--- ekg-20020414-new/docs/vars.txt	Mon Apr 15 21:49:08 2002
***************
*** 166,169 ****
--- 166,176 ----
  	,,use_proxy''.
  
+ qery_commands
+ 	typ: bool
+ 	domy¶lna warto¶æ: false
+ 
+ 	mo¿liwo¶æ wydawania d³ugich poleceñ podczas rozmowy z u¿ytkownikiem
+ 	bez poprzedzania ich znakiem '/'.
+ 
  quit_reason
  	typ: tekst
diff -r -C 2 ekg-20020414/src/commands.c ekg-20020414-new/src/commands.c
*** ekg-20020414/src/commands.c	Thu Apr 11 13:16:53 2002
--- ekg-20020414-new/src/commands.c	Mon Apr 15 21:42:14 2002
***************
*** 1410,1413 ****
--- 1410,1420 ----
  						tmp = "(...)";
  					my_printf("variable", v->name, tmp);
+ 				} else if (v->type == VAR_BOOL) {
+ 					char * tmp;
+ 					if (*(int*)(v->ptr))
+ 						tmp = "true";
+ 					else
+ 						tmp = "false";
+ 					my_printf("variable", v->name, tmp);
  				} else {
  					my_printf("variable", v->name, (!v->display) ? "(...)" : itoa(*(int*)(v->ptr)));
***************
*** 2007,2018 ****
  	int (*last_abbr)(char *, char **) = NULL;
  	int abbrs = 0;
  
  	if (query_nick && *line != '/') {
! 		char *params[] = { query_nick, line, NULL };
  
! 		if (strcmp(line, ""))
! 			command_msg("chat", params);
  
! 		return 0;
  	}
  	
--- 2014,2043 ----
  	int (*last_abbr)(char *, char **) = NULL;
  	int abbrs = 0;
+ 	int correct_command = 0;
  
  	if (query_nick && *line != '/') {
! 	
! 		/* wykrywanie przypadkowo wpisanych poleceñ */
! 		if (config_query_commands) {
! 			for (c = commands; c->name; c++) {
! 				int l=strlen(c->name);
! 				if ((l>2) && (strncasecmp(line, c->name, l)==0)) {
! 					if ((line[l]=='\0') || (isspace(line[l]))) {
! 					    correct_command=1;
! 					    break;
! 					}
! 				}
! 			}		
! 		}
! 
! 		if (!correct_command) {
! 	
! 			char *params[] = { query_nick, line, NULL };
  
! 			if (strcmp(line, ""))
! 				command_msg("chat", params);
  
! 			return 0;
! 		}
  	}
  	
diff -r -C 2 ekg-20020414/src/stuff.c ekg-20020414-new/src/stuff.c
*** ekg-20020414/src/stuff.c	Tue Apr  9 13:50:47 2002
--- ekg-20020414-new/src/stuff.c	Mon Apr 15 21:46:18 2002
***************
*** 103,106 ****
--- 103,107 ----
  char *config_away_reason = NULL;
  int config_random_reason = 0;
+ int config_query_commands = 0;
  
  /*
diff -r -C 2 ekg-20020414/src/stuff.h ekg-20020414-new/src/stuff.h
*** ekg-20020414/src/stuff.h	Tue Apr  9 13:50:47 2002
--- ekg-20020414-new/src/stuff.h	Mon Apr 15 21:11:40 2002
***************
*** 101,104 ****
--- 101,105 ----
  char *config_away_reason;
  int config_random_reason;
+ int config_query_commands;
  
  char *home_dir;
diff -r -C 2 ekg-20020414/src/vars.c ekg-20020414-new/src/vars.c
*** ekg-20020414/src/vars.c	Thu Mar 28 13:40:45 2002
--- ekg-20020414-new/src/vars.c	Mon Apr 15 21:02:26 2002
***************
*** 64,67 ****
--- 64,68 ----
  	variable_add("random_reason", VAR_INT, 1, &config_random_reason, NULL);
  	variable_add("quit_reason", VAR_STR, 1, &config_quit_reason, NULL);
+ 	variable_add("query_commands", VAR_BOOL, 1, &config_query_commands, NULL);
  	variable_add("sms_away", VAR_BOOL, 1, &config_sms_away, NULL);
  	variable_add("sms_max_length", VAR_INT, 1, &config_sms_max_length, NULL);
