ascent@linux.pl

diff -urN ekg-2001100102/Makefile.in ekg-ascent/Makefile.in
--- ekg-2001100102/Makefile.in	Mon Oct  1 01:02:40 2001
+++ ekg-ascent/Makefile.in	Mon Oct  1 10:41:37 2001
@@ -7,7 +7,7 @@
 NAME = @name@
 VERSION = @version@
 
-CFLAGS = -Wall -O3 -ggdb $(INCLUDES)
+CFLAGS = -Wall -O3 -ggdb @cflags@ $(INCLUDES)
 OBJS = libgg.o libgg_search.o stuff.o commands.o events.o themes.o vars.o dynstuff.o
 
 all:	ekg
diff -urN ekg-2001100102/configure ekg-ascent/configure
--- ekg-2001100102/configure	Mon Oct  1 01:02:40 2001
+++ ekg-ascent/configure	Mon Oct  1 10:43:50 2001
@@ -13,6 +13,7 @@
 #
 
 cc=gcc
+cflags=
 prefix=/usr/local
 includes=
 libs=
@@ -118,6 +119,36 @@
 echo "mhm."
 
 #
+# mozemy uzyc -pipe?
+#
+
+echo -n "sprawdzimy, czy kompilator obsluguje opcje -pipe... "
+cat > $tmpfile_c << EOF
+int main() {
+	return 7;
+}
+EOF
+
+if ! $cc $tmpfile_c -pipe -o $tmpfile > /dev/null 2>&1; then
+  rm -f $tmpfile $tmpfile_c
+  echo "nie."
+  exit 1
+fi
+
+$tmpfile
+
+if [ "$?" != "7" ]; then
+  rm -f $tmpfile $tmpfile_c
+  echo "jakos dziwnie??"
+  exit 1
+fi
+
+rm -f $tmpfile $tmpfile_c
+cflags="$cflags -pipe"
+
+echo "taaak."
+
+#
 # endiany.
 #
 
@@ -242,7 +273,7 @@
 
 rm -f $makefile
 echo "# plik wygenerowany automatyczny przez skrypt ,,configure''. nie zmieniaæ." > $makefile
-if ! sed -e "s|@includes@|${includes}|" -e "s|@libs@|${libs}|" -e "s|@prefix@|${prefix}|" -e "s|@name@|${name}|" -e "s|@version@|${version}|" -e "s|@cc@|${cc}|" < ${makefile}.in | grep -v '# \$\I\d' >> $makefile; then
+if ! sed -e "s|@includes@|${includes}|" -e "s|@libs@|${libs}|" -e "s|@prefix@|${prefix}|" -e "s|@name@|${name}|" -e "s|@version@|${version}|" -e "s|@cc@|${cc}|" -e "s|@cflags@|${cflags}|" < ${makefile}.in | grep -v '# \$\I\d' >> $makefile; then
   echo "b³±d sed'a."
   rm -f $makefile
   exit 1
