phantom@linux.bydg.org


diff -ruN ekg-20010822/CVS/Entries ekg-unofficial/CVS/Entries
--- ekg-20010822/CVS/Entries	Thu Jan  1 01:00:00 1970
+++ ekg-unofficial/CVS/Entries	Fri Aug 24 10:45:46 2001
@@ -0,0 +1,15 @@
+/TODO/1.1/Wed Aug 22 22:31:59 2001//
+/commands.h/1.1.1.1/Wed Aug 22 16:31:20 2001//
+/events.h/1.1.1.1/Wed Aug 22 16:31:20 2001//
+/gg.c/1.2/Wed Aug 22 22:25:30 2001//
+/libgg.c/1.2/Wed Aug 22 22:25:32 2001//
+/libgg.h/1.2/Wed Aug 22 22:25:34 2001//
+/stuff.c/1.2/Wed Aug 22 22:25:36 2001//
+/stuff.h/1.2/Wed Aug 22 22:25:37 2001//
+D/docs////
+/README.unofficialpatches/1.1/Thu Aug 23 22:13:59 2001//
+/commands.c/1.3/Thu Aug 23 22:15:22 2001//
+/events.c/1.3/Thu Aug 23 22:15:23 2001//
+/up_logging.h/1.1/Thu Aug 23 22:05:28 2001//
+/Makefile/1.4/Fri Aug 24 08:45:42 2001//
+/up_logging.c/1.2/Fri Aug 24 08:42:26 2001//
diff -ruN ekg-20010822/CVS/Repository ekg-unofficial/CVS/Repository
--- ekg-20010822/CVS/Repository	Thu Jan  1 01:00:00 1970
+++ ekg-unofficial/CVS/Repository	Thu Aug 23 00:33:31 2001
@@ -0,0 +1 @@
+ekg-unofficial
diff -ruN ekg-20010822/CVS/Root ekg-unofficial/CVS/Root
--- ekg-20010822/CVS/Root	Thu Jan  1 01:00:00 1970
+++ ekg-unofficial/CVS/Root	Thu Aug 23 00:33:31 2001
@@ -0,0 +1 @@
+:pserver:phantom@127.0.0.1:/home/cvs/CVSROOT
diff -ruN ekg-20010822/Makefile ekg-unofficial/Makefile
--- ekg-20010822/Makefile	Wed Aug 22 23:26:58 2001
+++ ekg-unofficial/Makefile	Fri Aug 24 10:45:42 2001
@@ -1,7 +1,9 @@
-# $Id$
+# $Id$
+
+UP_OBJ = up_logging.o
 
 CC = gcc -Wall -O3 -I/usr/include/readline
-OBJS = libgg.o stuff.o commands.o events.o
+OBJS = libgg.o stuff.o commands.o events.o $(UP_OBJ)
 LIBS = -lreadline -lncurses
 DATE = `date +'%Y%m%d'`
 
diff -ruN ekg-20010822/README ekg-unofficial/README
--- ekg-20010822/README	Wed Aug 22 02:06:11 2001
+++ ekg-unofficial/README	Thu Jan  1 01:00:00 1970
@@ -1,30 +0,0 @@
-
-              Motto: ,,Bunkrów nie ma, ale te¿ jest zajebi¶cie''
-
-INSTALACJA
-
-    utworzyæ plik ~/.gg/config zawieraj±cy:
-
-    uin <numerek>
-    password <has³o>
-
-    i odpaliæ to co¶. proste? proste. od razu mo¿na sobie dopisaæ mój
-    numerek 535333 (tak w ramach kryptoreklamy ;))
-
-CZEGO NIE MA
-
-    komunikacji miêdzy klientami, zak³adania userów, wysy³ania smsów.
-
-LICENCJA
-
-    blah! to jest jeszcze bardzo-bardzo-bardzo beta. nie rozprowadzaæ tego,
-    bo du¿o mo¿e siê zmieniæ. jedynie API libgg powinno byæ sta³e, wiêc mo¿na
-    ju¿ powoli siê przymierzaæ do X-owych frontendów. 
-
-ELEET GREETZ
-
-    Robert Wo¼ny (vinil@atman.pl)
-    Darek Jackowski (ascent@linux.pl)
-    Leszek Urbañski (tygrys@moo.pl)
-    Kuba Jermak (kooba@irc.pl)
-
diff -ruN ekg-20010822/README.unofficialpatches ekg-unofficial/README.unofficialpatches
--- ekg-20010822/README.unofficialpatches	Thu Jan  1 01:00:00 1970
+++ ekg-unofficial/README.unofficialpatches	Fri Aug 24 00:13:59 2001
@@ -0,0 +1,5 @@
+-- Fri Aug 24 00:12:44 CEST 2001 
+
+	* new commands ( log, awaylogging )
+	* some text changes (zdzichu's idea ;))
+
diff -ruN ekg-20010822/commands.c ekg-unofficial/commands.c
--- ekg-20010822/commands.c	Wed Aug 22 13:09:26 2001
+++ ekg-unofficial/commands.c	Fri Aug 24 00:15:22 2001
@@ -1,4 +1,4 @@
-/* $Id$ */
+/* $Id$ */
 
 #include <stdio.h>
 #include <unistd.h>
@@ -10,6 +10,11 @@
 #include "stuff.h"
 #include "commands.h"
 
+/*UP*/
+#include "up_logging.h"
+/*UP*/
+
+
 /*
  * jaka¶ malutka lista tych, do których by³y wysy³ane wiadomo¶ci.
  */
@@ -53,6 +58,9 @@
 	{ "send", "u?", command_send, "<numer/alias> <wiadomo¶æ>", "Wysy³a wiadomo¶æ do podanego u¿ytkownika", "" },
 	{ "quit", "", command_quit, "", "Wychodzi z programu", "" },
 	{ "unignore", "i", command_ignore, "<numer/alias>", "Usuwa z listy ignorowanych osób", "" },
+/*UP*/  { "log", "?", up_command_log, "0/1/2", "Wlacza/wylacza logoanie wiadomosci",""},
+/*UP*/  { "awaylogging","",up_command_awaylogging,"Wlacza/wylacza logowanie wiadomosci gdy jestes `zajety`","" },
+/*UP*/  { "play","",up_command_play,"Odczytanie zapisanych wiadomosci",""},
 	{ NULL, NULL, NULL, NULL, NULL }
 };
 
@@ -255,9 +263,11 @@
 	if (!strcasecmp(name, "away")) {
 		gg_change_status(sock, GG_STATUS_BUSY);
 		my_printf(PROMPT "Zmieniono stan na zajêty\n");
+		up_away=1;
 	} else {
 		gg_change_status(sock, GG_STATUS_AVAIL);
 		my_printf(PROMPT "Zmieniono stan na dostêpny\n");
+		up_away=0;
 	}
 	
 	return 0;
@@ -373,10 +383,10 @@
 		tmp = "";
 		switch (u->status) {
 			case GG_STATUS_AVAIL:
-				tmp = "\033[1;33m(dostêpny)\033[0m";
+				tmp = "\033[1;33m(otwarty na propozycje)\033[0m";
 				break;
 			case GG_STATUS_NOT_AVAIL:
-				tmp = "\033[0;31m(niedostêpny)\033[0m";
+				tmp = "\033[0;31m(zonaty)\033[0m";
 				break;
 			case GG_STATUS_BUSY:
 				tmp = "\033[1;36m(zajêty)\033[0m";
diff -ruN ekg-20010822/commands.h ekg-unofficial/commands.h
--- ekg-20010822/commands.h	Wed Aug 22 02:06:38 2001
+++ ekg-unofficial/commands.h	Wed Aug 22 18:31:20 2001
@@ -1,4 +1,4 @@
-/* $Id$ */
+/* $Id$ */
 
 #ifndef __COMMANDS_H
 #define __COMMANDS_H
Binary files ekg-20010822/commands.o and ekg-unofficial/commands.o differ
diff -ruN ekg-20010822/docs/CVS/Entries ekg-unofficial/docs/CVS/Entries
--- ekg-20010822/docs/CVS/Entries	Thu Jan  1 01:00:00 1970
+++ ekg-unofficial/docs/CVS/Entries	Thu Aug 23 00:33:44 2001
@@ -0,0 +1,5 @@
+/README/1.1/Wed Aug 22 22:32:04 2001//
+/ggregister/1.1/Wed Aug 22 22:32:08 2001//
+/ggsearch/1.1/Wed Aug 22 22:32:13 2001//
+/protocol.txt/1.1/Wed Aug 22 22:32:15 2001//
+D
diff -ruN ekg-20010822/docs/CVS/Repository ekg-unofficial/docs/CVS/Repository
--- ekg-20010822/docs/CVS/Repository	Thu Jan  1 01:00:00 1970
+++ ekg-unofficial/docs/CVS/Repository	Thu Aug 23 00:33:42 2001
@@ -0,0 +1 @@
+ekg-unofficial/docs
diff -ruN ekg-20010822/docs/CVS/Root ekg-unofficial/docs/CVS/Root
--- ekg-20010822/docs/CVS/Root	Thu Jan  1 01:00:00 1970
+++ ekg-unofficial/docs/CVS/Root	Thu Aug 23 00:33:42 2001
@@ -0,0 +1 @@
+:pserver:phantom@127.0.0.1:/home/cvs/CVSROOT
diff -ruN ekg-20010822/docs/ggregister ekg-unofficial/docs/ggregister
--- ekg-20010822/docs/ggregister	Wed Aug 22 23:24:49 2001
+++ ekg-unofficial/docs/ggregister	Thu Aug 23 00:32:08 2001
@@ -1,5 +1,5 @@
 #!/usr/bin/perl
-# $Id$
+# $Id$
 
 use IO::Socket;
 
diff -ruN ekg-20010822/docs/ggsearch ekg-unofficial/docs/ggsearch
--- ekg-20010822/docs/ggsearch	Tue Jan 23 13:51:42 2001
+++ ekg-unofficial/docs/ggsearch	Thu Aug 23 00:32:13 2001
@@ -1,5 +1,5 @@
 #!/usr/bin/perl
-# $Id$
+# $Id$
 
 use IO::Socket;
 $/ = "\r\n";
diff -ruN ekg-20010822/docs/protocol.txt ekg-unofficial/docs/protocol.txt
--- ekg-20010822/docs/protocol.txt	Tue Aug 21 23:45:30 2001
+++ ekg-unofficial/docs/protocol.txt	Thu Aug 23 00:32:15 2001
@@ -6,7 +6,7 @@
 
 : protogó³ gadu-gadu 4.x
 | (c) copyright 2001 by wojtek kaniewski <wojtekka@irc.pl>
-| $Id$
+| $Id$
 `------------------------------------------- ----- ---- -- -
 
 --- 0) disclaimer ---------------------------------------------------------
diff -ruN ekg-20010822/events.c ekg-unofficial/events.c
--- ekg-20010822/events.c	Wed Aug 22 14:55:56 2001
+++ ekg-unofficial/events.c	Fri Aug 24 00:15:23 2001
@@ -1,4 +1,4 @@
-/* $Id$ */
+/* $Id$ */
 
 #include <stdio.h>
 #include <unistd.h>
@@ -42,6 +42,7 @@
 
 	cp_to_iso(e->event.msg.message);
 	my_printf(PROMPT "\007Wiadomo¶æ od %s\n%s\n", format_user(e->event.msg.sender), e->event.msg.message);
+/* UP */ log_msg(e->event.msg.sender,e->event.msg.message);
 }
 
 void handle_ack(struct gg_event *e)
@@ -66,9 +67,9 @@
 		if ((u = find_user(n->uin, NULL)))
 			u->status = n->status;
 		if (n->status == GG_STATUS_AVAIL)
-			my_printf(PROMPT "%s dostêpny\n", format_user(n->uin));
+			my_printf(PROMPT "%s jest otwarty na wszelakie propozycje\n", format_user(n->uin));
 		else if (n->status == GG_STATUS_BUSY)
-			my_printf(PROMPT "%s zajêty\n", format_user(n->uin));
+			my_printf(PROMPT "%s jest zajêty\n", format_user(n->uin));
 		n++;
 	}
 }
@@ -84,11 +85,11 @@
 		u->status = e->event.status.status;
 
 	if (e->event.status.status == GG_STATUS_AVAIL)
-		my_printf(PROMPT "%s jest ju¿ dostêpny\n", format_user(e->event.status.uin));
+		my_printf(PROMPT "%s jest otwarty na propozycje wszelakie\n", format_user(e->event.status.uin));
 	else if (e->event.status.status == GG_STATUS_BUSY)
 		my_printf(PROMPT "%s jest zajêty\n", format_user(e->event.status.uin));
 	else if (e->event.status.status == GG_STATUS_NOT_AVAIL)
-		my_printf(PROMPT "%s jest niedostêpny\n", format_user(e->event.status.uin));
+		my_printf(PROMPT "%s ozenil sie\n", format_user(e->event.status.uin));
 }
 
 void handle_event()
diff -ruN ekg-20010822/events.h ekg-unofficial/events.h
--- ekg-20010822/events.h	Wed Aug 22 02:06:43 2001
+++ ekg-unofficial/events.h	Wed Aug 22 18:31:20 2001
@@ -1,4 +1,4 @@
-/* $Id$ */
+/* $Id$ */
 
 #ifndef __EVENTS_H
 #define __EVENTS_H
Binary files ekg-20010822/events.o and ekg-unofficial/events.o differ
diff -ruN ekg-20010822/gg.c ekg-unofficial/gg.c
--- ekg-20010822/gg.c	Wed Aug 22 12:49:04 2001
+++ ekg-unofficial/gg.c	Thu Aug 23 00:25:30 2001
@@ -1,4 +1,4 @@
-/* $Id$ */
+/* $Id$ */
 
 #include <stdio.h>
 #include <unistd.h>
diff -ruN ekg-20010822/libgg.c ekg-unofficial/libgg.c
--- ekg-20010822/libgg.c	Wed Aug 22 12:35:40 2001
+++ ekg-unofficial/libgg.c	Thu Aug 23 00:25:32 2001
@@ -1,4 +1,4 @@
-/* $Id$ */
+/* $Id$ */
 
 #include <arpa/inet.h>
 #include <sys/socket.h>
@@ -10,7 +10,17 @@
 #include <string.h>
 #include <stdarg.h>
 #include <pwd.h>
-#include <endian.h>
+
+#include <netdb.h>
+#include <sys/types.h>
+#include <netinet/in.h>
+#include <netinet/in_systm.h>
+#include <netinet/ip.h>
+#include <netinet/tcp.h>
+#include <netinet/ip_icmp.h>
+#include <ctype.h>
+
+/* #include <endian.h> */
 #include "libgg.h"
 
 int gg_debug_level = 0;
diff -ruN ekg-20010822/libgg.h ekg-unofficial/libgg.h
--- ekg-20010822/libgg.h	Wed Aug 22 12:35:57 2001
+++ ekg-unofficial/libgg.h	Thu Aug 23 21:18:15 2001
@@ -1,4 +1,4 @@
-/* $Id$ */
+/* $Id$ */
 
 #ifndef __LIBGG_H
 #define __LIBGG_H
Binary files ekg-20010822/libgg.o and ekg-unofficial/libgg.o differ
diff -ruN ekg-20010822/stuff.c ekg-unofficial/stuff.c
--- ekg-20010822/stuff.c	Wed Aug 22 13:04:06 2001
+++ ekg-unofficial/stuff.c	Thu Aug 23 23:25:28 2001
@@ -1,4 +1,4 @@
-/* $Id$ */
+/* $Id$ */
 
 #include <stdio.h>
 #include <stdlib.h>
diff -ruN ekg-20010822/stuff.h ekg-unofficial/stuff.h
--- ekg-20010822/stuff.h	Wed Aug 22 12:48:52 2001
+++ ekg-unofficial/stuff.h	Thu Aug 23 00:25:37 2001
@@ -1,4 +1,4 @@
-/* $Id$ */
+/* $Id$ */
 
 #ifndef __STUFF_H
 #define __STUFF_H
Binary files ekg-20010822/stuff.o and ekg-unofficial/stuff.o differ
diff -ruN ekg-20010822/up_logging.c ekg-unofficial/up_logging.c
--- ekg-20010822/up_logging.c	Thu Jan  1 01:00:00 1970
+++ ekg-unofficial/up_logging.c	Fri Aug 24 10:42:26 2001
@@ -0,0 +1,110 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <time.h>
+#include <errno.h>
+#include <sys/stat.h>
+#include "libgg.h"
+#include "commands.h"
+#include "stuff.h"
+#include "up_logging.h"
+
+int up_logging=1;
+int up_away_logging=1;
+
+
+COMMAND(up_command_awaylogging) {
+	if (up_away_logging)
+		{
+		my_printf(PROMPT "Logowanie wiadomosci zostalo wylaczone.\n");
+		up_away_logging=0;
+		}else{
+		my_printf(PROMPT "Logowanie wiadomosci zostalo wlaczone.\n");
+		up_away_logging=1;
+		}
+return 0;
+}
+
+
+COMMAND(up_command_play) {
+	my_printf(ERROR "Not implemented yet.\n");
+return 0;
+}
+
+COMMAND(up_command_log) {
+	if ((!params[0])||((*params[0]!='0')&&\
+           (*params[0]!='1')&&(*params[0]!='2')))
+		{
+		my_printf(PROMPT "Aktualnie logowanie jest ustawione na:%i\n",up_logging);
+		my_printf(ERROR "Zle wywolanie funckji `log`\n");
+		my_printf(ERROR "parametrem do funkcji 'log'\n");
+		my_printf(ERROR "moga byc cyfry od 0 do 2 oznaczajace kolejno:\n");
+		my_printf(ERROR "0 - brak logowania,\n");
+ 		my_printf(ERROR "1 - logowanie do pliku gg.log,\n");
+		my_printf(ERROR "2 - logowanie do pliku (uin:alias).log\n");
+		return 0;
+		}
+up_logging=atoi(params[0]);
+return 0;
+}
+
+FILE *aopen(char *path) {
+int fd;
+FILE *o=NULL;
+mode_t prv_umask;
+        prv_umask=umask(0);
+        fd=open(path,O_CREAT|O_RDWR|O_NOFOLLOW,S_IRUSR|S_IWUSR);
+        if (errno!=0) o=fdopen(fd,"a+");
+        umask(prv_umask);
+        return o;
+}
+
+void log_msg(uin_t sender,char *message) {
+FILE  *logfile;
+char  *logfilename;
+char  *acttime;
+char  *csender=(char*)malloc(64);
+struct userlist *u = find_user(sender, NULL);
+time_t secs=time(NULL);
+	if (u)
+		{
+		snprintf(csender,63,"(%u:%s)",sender, u->comment);
+		} else {
+		snprintf(csender,63,"%u",sender);
+		}
+	acttime=ctime(&secs);
+	acttime[strlen(acttime)-1]=0x00; /* brutal :) but it works ;> */
+	switch(up_logging) {
+	case 0 : { break;} ;
+	case 1 : {
+		   if ((logfile=aopen("gg.log"))!=NULL)
+			{
+		   	fprintf(logfile,"<%s:%s> %s\n",acttime,csender,message);
+		   	fclose(logfile);
+		   	}
+		   break; 
+	         };
+	case 2 : {
+		   logfilename=(char*)malloc(128);
+		   snprintf(logfilename,127,"%s.log",csender);
+		   if ((logfile=aopen(logfilename))!=NULL)
+			{
+	           	fprintf(logfile,"<%s:%s> %s\n",acttime,csender,message);
+	           	fclose(logfile);
+			}
+		   free(logfilename);
+		   break; 
+		 };
+	default: ;
+	}
+	if (up_away_logging&&up_away)
+		{
+			if((logfile=aopen("gg.away"))!=NULL)
+				{
+				fprintf(logfile,"<%s:%s> %s\n",acttime,csender,message);
+				fclose(logfile);
+				}
+		}
+		return;
+}
+
diff -ruN ekg-20010822/up_logging.h ekg-unofficial/up_logging.h
--- ekg-20010822/up_logging.h	Thu Jan  1 01:00:00 1970
+++ ekg-unofficial/up_logging.h	Fri Aug 24 00:05:28 2001
@@ -0,0 +1,7 @@
+int up_away;          /* away - on/off */
+
+void log_msg(uin_t sender, char *message); /* it will log message from
+                                              user in selected by
+                                              /log command mode */
+int up_command_log(), up_command_awaylogging(), up_command_play();
+
Binary files ekg-20010822/up_logging.o and ekg-unofficial/up_logging.o differ
